frontend/packages/dev-console/src/components/import/import-types.ts

```diff
@@ -9,10 +9,32 @@ export interface DeployImageFormProps {
   imageStreams?: FirehoseList;
 }
 
-export interface ImageStreamProps {
-  imageStreams: K8sResourceKind[];
+export type ImageStreamState = {
+  hasAccessToPullImage: boolean;
+  accessLoading: boolean;
+  loading: boolean;
+  hasCreateAccess: boolean;
+  selectedImageStream: K8sResourceKind | any;
+};
+export enum ImageStreamActions {
+  setAccessLoading = 'setAccessLoading',
+  setLoading = 'setLoading',
+  setSelectedImageStream = 'setSelectedImageStream',
+  setHasAccessToPullImage = 'setHasAccessToPullImage',
+  setHasCreateAccess = 'setHasCreateAccess',
+}
+export type ImageStreamAction =
+  | { type: ImageStreamActions.setAccessLoading; value: boolean }
+  | { type: ImageStreamActions.setLoading; value: boolean }
+  | { type: ImageStreamActions.setSelectedImageStream; value: K8sResourceKind | any }
```

Comment:
Isn't ```value``` is always of kind ```K8sResourceKind```.

Reply:
Yes It is, but it throws weird  lint error since we are using `|` statement. The first `setAccessLoading` statement is also picking up the `K8sResourceKind` | `boolean`, thats why i add any here to avoid lint error. but i agree, this needs to be fixed.

Reply:
Can we simply define the type like this 
```export type ImageStreamAction = { type: ImageStreamActions; value: boolean | K8sResourceKind };```